TestTChannelTracerDepth2 flaky

```
--- FAIL: TestTChannelTracerDepth2 (0.01s)
	Error Trace:	tracer_test.go:235
			tracer_test.go:231
	Error:      	Not equal: 
	            	expected: 4
	            	received: 3
	Messages:   	generates inbound and outband spans
```

Looks like this one is also happening with TestTChannelTracerDepth.

```
--- FAIL: TestTChannelTracer (0.01s)
	Error Trace:	tracer_test.go:178
			tracer_test.go:174
	Error:      	Not equal: 
	            	expected: 2
	            	received: 1
	Messages:   	generates inbound and outband spans
```

Took a look at the code, and this might be happening due to a race condition with finishing the opentracing span on TChannel inbounds.

The inbound finishes/closes the span __after__ it's finished sending the response back.  
https://github.com/uber/tchannel-go/blob/dev/fragmenting_writer.go#L272
https://github.com/uber/tchannel-go/blob/dev/inbound.go#L384

this means that there's a possible race where our assertion happens before the inbound is done cleaning up the tracer span.

To fix this we'll probably need to rework how this test works.

--- a/transport/tracer_test.go
+++ b/transport/tracer_test.go
@@ -71,7 +71,7 @@ func (h handler) echoEcho(ctx context.Context) error {
 
 func (h handler) createContextWithBaggage(tracer opentracing.Tracer) (context.Context, func()) {
 	ctx := context.Background()
-	ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
 
 	span := tracer.StartSpan("test")
 	// no defer span.Finish()
@@ -157,6 +157,10 @@ func TestHTTPTracer(t *testing.T) {
 func TestTChannelTracer(t *testing.T) {
 	tracer := mocktracer.New()
 	dispatcher := createTChannelDispatcher(tracer, t)
+	// Make this assertion at the end of the defer stack, when the channel has
+	// been shut down. This ensures that all message exchanges have been shut
+	// down, which means that all spans have been closed.
+	defer AssertDepth1Spans(t, tracer)
 
 	client := json.New(dispatcher.ClientConfig("yarpc-test"))
 	handler := handler{client: client, t: t}
@@ -170,8 +174,6 @@ func TestTChannelTracer(t *testing.T) {
 
 	err := handler.echo(ctx)
 	assert.NoError(t, err)
-
-	AssertDepth1Spans(t, tracer)
 }
 
 func AssertDepth1Spans(t *testing.T, tracer *mocktracer.MockTracer) {
@@ -207,13 +209,16 @@ func TestHTTPTracerDepth2(t *testing.T) {
 
 	err := handler.echoEcho(ctx)
 	assert.NoError(t, err)
-
 	AssertDepth2Spans(t, tracer)
 }
 
 func TestTChannelTracerDepth2(t *testing.T) {
 	tracer := mocktracer.New()
 	dispatcher := createTChannelDispatcher(tracer, t)
+	// Make this assertion at the end of the defer stack, when the channel has
+	// been shut down. This ensures that all message exchanges have been shut
+	// down, which means that all spans have been closed.
+	defer AssertDepth2Spans(t, tracer)
 
 	client := json.New(dispatcher.ClientConfig("yarpc-test"))
 	handler := handler{client: client, t: t}
@@ -227,8 +232,6 @@ func TestTChannelTracerDepth2(t *testing.T) {
 
 	err := handler.echoEcho(ctx)
 	assert.NoError(t, err)
-
-	AssertDepth2Spans(t, tracer)
 }
 
 func AssertDepth2Spans(t *testing.T, tracer *mocktracer.MockTracer) {

